iOS 7차 과제 - 데이터 연결 심화, UIGestureRecognizer

EunHee-Jeong opened the pull request:
## 📌 관련 이슈
<!-- 관련있는 이슈 번호(#000)을 적어주세요.
  해당 pull request merge와 함께 이슈를 닫으려면 
  closed #Issue_number를 적어주세요 -->
  closed #17 

## 📌 변경 사항 및 이유
<!-- 변경한 내용과 그 이유를 적어주세요. -->
<img src="https://user-images.githubusercontent.com/70689381/146896646-ddb27ffc-3177-4d65-95d1-5b377a3d1d7a.gif" width="250">

## 📌 PR Point
<!-- 리뷰어 분들이 집중적으로 보셨으면 하는 내용을 적어주세요 -->
- HomeVC에 UIGestureRecognizerDelegate를 채택해주고 UITableViewDataSource에서 제스처 인식과 데이터 전달을 해줬습니다...!
- ~화면 전환될 때 데이터가 전달되는 과정이 너무 잘 보이는데 왜 그럴까요??~
    - 해결 완료: 데이터 전달 방식 변경 (클로저 ➡️ 프로퍼티)
<img src="https://user-images.githubusercontent.com/70689381/147618798-051529f9-321f-4619-b32f-ed17e4cf2b72.gif" width="250">

## 📌 참고 사항
<!-- 참고할 사항이 있다면 적어주세요. -->

Commit:
✨[FEAT] imageView에 tap gesture recognizer 연결하기 완료 (#17)

Commit:
➕[ADD] HomeTapVC.swift 파일 생성과 표시할 스토리보드 추가 (#17)

Commit:
🍱[ADD] Back Btn용 이미지 에셋 추가 (#17)

Commit:
💄[MOD] HomeTabVC UI 구성 (#17)

Commit:
✨[FEAT] present 화면 전환과 데이터 전달하기 완료 (#17)

Commit:
✨[FEAT] backBtn 눌렀을 때 이전 화면으로 돌아가기 완료 (#17)

Commit:
💄[MOD] titleLabel 글씨 다 보이게 수정 (#17)

Text->Attributed, Lines->2, Line Break->Word Wrap, Tooltip에서 Shows Expansion Text 체크해주기

Taehyeon-Kim commented on seminar-homework-1/seminar-homework-1/Source/ViewController/Tabbars/HomeTapVC.swift:
```diff
@@ -0,0 +1,37 @@
+import UIKit
+
+class HomeTapVC: UIViewController {
+    
```
공백 제거..!

Taehyeon-Kim commented on seminar-homework-1/seminar-homework-1/Source/ViewController/Tabbars/HomeTapVC.swift:
```diff
@@ -0,0 +1,37 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        imageTintColorSettings()
+
```
여기도.. 공백 제거..!

Taehyeon-Kim commented on seminar-homework-1/seminar-homework-1/Source/ViewController/Tabbars/HomeTapVC.swift:
```diff
@@ -0,0 +1,37 @@
+    }
+    
+    func imageTintColorSettings() {
+        let image = UIImage(named: "locationIcon.png")?.withRenderingMode(.alwaysTemplate)
```
이미지 컬러 때문에 쓰셨군요 디테일 👍

Taehyeon-Kim commented on seminar-homework-1/seminar-homework-1/Source/ViewController/Tabbars/HomeVC.swift:
```diff
@@ -138,9 +141,28 @@ extension HomeVC: UITableViewDataSource {
     
+    @objc func tapView(gestureRecognizer: UIGestureRecognizer) {
+        
+        guard let nextVC = self.storyboard?.instantiateViewController(withIdentifier: HomeTapVC.identifier) as? HomeTapVC else {return}
+        nextVC.modalPresentationStyle = .fullScreen
+        present(nextVC, animated: true) {
+            nextVC.videoImageView.image = UIImage(named: "wesoptiOSPart")
+            nextVC.titleLabel.text = "1차 세미나 : iOS 컴포넌트 이해, XCode 기본 사용법, View 화면 전환"
+            nextVC.descriptionLabel.text = "WE SOPT ・조회수 100만회 ・ 3주 전"
+        }
```
```suggestion
        guard let nextVC = self.storyboard?.instantiateViewController(withIdentifier: HomeTapVC.identifier) as? HomeTapVC else {return}
        nextVC.modalPresentationStyle = .fullScreen
        // 예시
        nextVC.setData(image:_, title:_, descripion:_)
        present(nextVC, animated: true) 
```

Taehyeon-Kim replied:
데이터가 늦게 표시되는 이유는 아무래도 클로저에서 작업을 처리해주고 있어서인 것 같아요.
흐름상 present이후에 데이터를 전달하고 있죠.
순서만 조금 변경해주면 될 것 같아요..!

EunHee-Jeong replied:
와 짱 빠르다...!!!
그렇다면 setData를 CVC에서 만들어두고 ➡️ VC에서 넘겨주라는 말씀이신가여...??

Taehyeon-Kim replied:
그러면 될 것 같은데요..!

EunHee-Jeong replied:
이해완료! 👍

Taehyeon-Kim reviewed (approved):
LZTM 👍

Commit:
✅[CHORE] 금잔디 리뷰 반영 - 데이터 전달 부분 코드 수정 (#17)

프로퍼티를 이용한 데이터 전달에서는 넘어올 값을 세팅해주는 함수를 viewDidLoad()에서 호출해줘야 한다!!!! (그렇지 않으면 VC의 초기화 작업 전에 메서드가 먼저 호출 되어버려 오류가 발생하게 됨)

Commit:
💄[MOD] Initial VC 변경과 주석 추가 (#17)
